pulp_rpm/app/viewsets/package.py

```diff
@@ -71,3 +71,50 @@ class PackageViewSet(SingleArtifactContentUploadViewSet):
         ],
         "queryset_scoping": {"function": "scope_queryset"},
     }
+
+    @extend_schema(
+        description="Trigger an asynchronous task to create an RPM package,"
+        "optionally create new repository version.",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    def create(self, request):
+        # normal case
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        sign_package = serializer.validated_data.pop("sign_package")
+        if sign_package is not True:
+            return super().create(request)
+
+        # signing case invariants
+        request.data.pop("file")
+        request.data.pop("sign_package")
+        temp_uploaded_file = serializer.validated_data.get("file")
+        signing_service_pk = serializer.validated_data.pop("signing_service_pk")
+        signing_fingerprint = serializer.validated_data.pop("signing_fingerprint")
+
+        # dispatch signing task
+        pulp_temp_file = PulpTemporaryFile(file=temp_uploaded_file.temporary_file_path())
+        pulp_temp_file.save()
+        task_args = {
+            "app_label": self.queryset.model._meta.app_label,
+            "serializer_name": serializer.__class__.__name__,
+            "signing_service_pk": signing_service_pk,
+            "signing_fingerprint": signing_fingerprint,
+            "temporary_file_pk": pulp_temp_file.pk,
+        }
+        task_payload = {k: v for k, v in request.data.items()}
+        task_exclusive = [
+            item
+            for item in (serializer.validated_data.get(key) for key in ("upload", "repository"))
+            if item
+        ]
```

Comment:
`[serializer.validated_data.get("upload"), serializer.validated_data.get("repository")]` might be simpler